bigcode_eval/utils.py

```diff
@@ -332,7 +334,8 @@ def complete_code(
                     gen_token_dict,
                 )
                 with open(intermediate_save_generations_path, "w") as fp:
-                    json.dump(code_gens, fp)
+                    intermediate_generations.extend(code_gens)
```

Comment:
if there are multiple saving steps, I think we'll add the same generations several times into `intermediate_generations`.
Also this list is also extended in `evaluator.py` https://github.com/bigcode-project/bigcode-evaluation-harness/blob/0754793a4ba22eaaeff2c5381a480dfc36700cbe/bigcode_eval/evaluator.py#L82

Reply:
you're right, made 2 changes in the new commit

1. instead of extending on `intermediate_generations` when saving which end up duplicating new generations, extend on a deepcopy instead which prevents duplications
2. instead of extending at evaluator, return `intermediate_generations.extend(code_gens)` in `complete_code`, which makes a bit more sense with this new logic. since we never mutate `intermediate_generations` when saving, this will return the non-duplicated generations